components/merkledb/src/proof_map_index/mod.rs

```diff
@@ -67,30 +68,23 @@ where
     }
 }
 
-/// A Merkelized version of a map that provides proofs of existence or non-existence for the map
-/// keys.
-///
-/// `ProofMapIndex` implements a Merkle Patricia tree, storing values as leaves.
-/// `ProofMapIndex` requires that keys implement the [`BinaryKey`] trait and
-/// values implement the [`BinaryValue`] trait.
-///
-/// [`BinaryKey`]: ../trait.BinaryKey.html
-/// [`BinaryValue`]: ../trait.BinaryValue.html
-pub struct ProofMapIndex<T: IndexAccess, K, V> {
+#[doc(hidden)]
+pub struct ProofMapIndexBase<T: IndexAccess, K, V, KeyMode: ToProofPath<K>> {
```

Comment:
I still think that having base a separate type rather than defining `ProofMapIndex` with a default `KeyMode` is unwarranted (could be solved later, I suppose). Right now, it could be fixed by introducing separate constructors for default / any `KeyMode`; with #1523 merged, it would be solved by having 2 separate methods in `AccessExt` for `ProofMapIndex`es with hashed / raw keys.